controllers/prep_handlers.go

```diff
@@ -393,24 +379,9 @@ func (r *ImageBasedUpgradeReconciler) prepStageWorker(ctx context.Context, ibu *
 			}
 			r.Log.Info("Successfully created precaching job")
 			r.PrepTask.Progress = "Successfully created precaching job"
+			r.Precache.PrecacheRequested = true
```

Comment:
I wonder if `r.Precahe.PrecacheRequested` is really needed? since if it reaches the end, it means the job was created successfully. If any error is raised during precache job creation, the prep will fail.

Reply:
Yeah! So it once the go routine is launched we may reconcile before the job is created. So it was just make sure we can safely check the job status only once the job is launched. 

But you're right...this is pretty ugly. Will refactor this

Reply:
Dropped PreacheRequested. Should be a lot cleaner now